tests/components/brother/test_config_flow.py

```diff
@@ -103,97 +60,90 @@ async def test_invalid_hostname(hass: HomeAssistant) -> None:
     assert result["errors"] == {CONF_HOST: "wrong_host"}
 
 
-@pytest.mark.parametrize("exc", [ConnectionError, TimeoutError])
-async def test_connection_error(hass: HomeAssistant, exc: Exception) -> None:
+@pytest.mark.parametrize(
+    ("exc", "base_error"),
+    [
+        (ConnectionError, "cannot_connect"),
+        (TimeoutError, "cannot_connect"),
+        (SnmpError("SNMP error"), "snmp_error"),
+    ],
+)
+async def test_errors(
+    hass: HomeAssistant, exc: Exception, base_error: str, mock_brother_client: AsyncMock
+) -> None:
     """Test connection to host error."""
-    with (
-        patch("brother.Brother.initialize"),
-        patch("brother.Brother._get_data", side_effect=exc),
-    ):
-        result = await hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": SOURCE_USER}, data=CONFIG
-        )
-
-        assert result["errors"] == {"base": "cannot_connect"}
+    mock_brother_client.async_update.side_effect = exc
 
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}, data=CONFIG
+    )
 
-async def test_snmp_error(hass: HomeAssistant) -> None:
-    """Test SNMP error."""
-    with (
-        patch("brother.Brother.initialize"),
-        patch("brother.Brother._get_data", side_effect=SnmpError("error")),
-    ):
-        result = await hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": SOURCE_USER}, data=CONFIG
-        )
-
-        assert result["errors"] == {"base": "snmp_error"}
+    assert result["errors"] == {"base": base_error}
 
 
 async def test_unsupported_model_error(hass: HomeAssistant) -> None:
     """Test unsupported printer model error."""
     with (
-        patch("brother.Brother.initialize"),
-        patch("brother.Brother._get_data", side_effect=UnsupportedModelError("error")),
+        patch("homeassistant.components.brother.Brother.initialize"),
+        patch(
+            "homeassistant.components.brother.Brother._get_data",
+            side_effect=UnsupportedModelError("error"),
+        ),
     ):
```

Comment:
2 things, why isn't this patched in the config flow, and can't we use the new mock for this?

Reply:
This exception may occur in the `Brother.create` method, and this method is used to create an instance of the `Brother` class, I was unable to get AsyncMock to pass the exception as a `side_effect`. Something like this doesn't work
```python
mock_brother_client.create.side_effect = UnsupportedModelError("error")
```
Maybe there is a better way to achieve this?

Reply:
I'm just going to throw some ideas, I will have some time later to actually test some
```py
mock_client.create.return_value.side_effect = UnsupportedModelError("error")
mock_client.create.return_value = UnsupportedModelError("error")
```

Reply:
Have you tried something like https://github.com/home-assistant/core/blob/f23419ed3504e189399bff7aa948ff0e710e88ee/tests/components/shelly/test_config_flow.py#L867-L869

Reply:
Yes, this simplifies tests a bit, but I still can't force a fixture mock to raise an exception for the `create()` method.